### PR TITLE
Remove sqlite and postgres from rest api feature

### PR DESCRIPTION
**Pull request by vaporos**
This removes sqlite and postgres from the sdk's rest-api-actix-web-3
feature, which will make it possible to conditionally use the sqlite or
postgresql when using the rest api.

This commit makes BatchList pub, because without pub, it would be
necessary to put a conditional of the underlying implementations being
enabled; it seems cleaner to simply make it pub since it does not seem
intentional to prevent creation of BatchList (which itself is pub).

Features are added to griddle for sqlite/postgres to pull in the
associated sdk features.

### CODE REVIEW

File: sdk/Cargo.toml [hunk 1]
```diff
@@ -129,18 +129,16 @@ schema-store-postgres = ["postgres"]
 schema-store-sqlite = ["sqlite"]
 track-and-trace = ["base64"]
 batch-processor = ["batch-store", "backend", "log", "uuid"]
-batch-store = []
+batch-store = ["chrono"]
```
Comment (peterschwarz) [inline sdk/Cargo.toml:132]: This change seems unrelated

Comment (vaporos) [reply sdk/Cargo.toml:132]: It is related, because with the change chrono is no longer pulled into the batch-store non-diesel code via sqlite/postgres implementation if they are disabled.

**Review by dnewh — APPROVED** [review]

**Review by peterschwarz — DISMISSED** [review]

File: griddle/src/main.rs [hunk 2]
```diff
@@ -47,11 +50,20 @@ fn log_format(
     write!(w, "{}", record.args(),)
 }
 
+#[cfg(not(any(feature = "database-postgres", feature = "database-sqlite")))]
+fn griddle_store_state(_db_url: &str) -> Result<StoreState, Error> {
+    Err(Error::from_message(
+        "no database feature was enabled durinig compilation",
```
Comment (dnewh) [inline griddle/src/main.rs:56]: s/durinig/during

**Review by dnewh — CHANGES_REQUESTED** [review]

File: griddle/src/main.rs (hunk 2 above)
Comment (vaporos) [reply griddle/src/main.rs]: fixed

**Review by dnewh — APPROVED** [review]

**Review by peterschwarz — APPROVED** [review]
